### ISSUE TITLE
Release new component mounting method

### ISSUE THREAD

**Issue by bahmutov** [bahmutov/cypress-angular-unit-test#136]
Trying in https://github.com/bahmutov/cypress-angular-unit-test/tree/try-as-component-spec

- clone the repository
- `git checkout try-as-component-spec`
- Install dependencies with `npm i`
- `npm run cy:open`
- click on the component spec filename 

<img width="796" alt="Screen Shot 2020-05-06 at 10 13 52 AM" src="https://user-images.githubusercontent.com/2212006/81187624-4b4bca00-8f82-11ea-993f-fdb9a22948bf.png">

but hitting an error

<img width="753" alt="Screen Shot 2020-05-05 at 3 58 01 PM" src="https://user-images.githubusercontent.com/2212006/81110234-425aea00-8ee9-11ea-8b28-e240b4965765.png">

The test tries to import and bootstrap a module

```
import 'core-js/es7/reflect';
import { ApplicationRef, NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
import 'zone.js';
import { AppComponent } from '../../src/app/app.component';

// dynamic loading based on blog post
// https://blog.angularindepth.com/how-to-manually-bootstrap-an-angular-application-9a36ccf86429

@NgModule({
  declarations: [
    AppComponent
  ],
  imports: [
    BrowserModule
  ],
  providers: [],
  entryComponents: [AppComponent]
})
class AppModule {
  app: ApplicationRef;
  ngDoBootstrap(app: ApplicationRef) {
    debugger
    this.app = app;
  }
}

/* eslint-env mocha */
/* global cy */
describe('AppComponent', () => {
  beforeEach(() => {
    const html = `
      <head>
        <meta charset="UTF-8">
      </head>
      <body>
        <app-root></app-root>
      </body>
    `;
    const document = (cy as any).state('document');
    document.write(html);
    document.close();

    cy.get('app-root').then(el$ => {
      platformBrowserDynamic()
        .bootstrapModule(AppModule)
        .then(function (moduleRef) {
          moduleRef.instance.app.bootstrap(AppComponent, el$.get(0));
        });
    });
  });
})
```

**Comment by bahmutov** [comment]
Seems upgrading to ng9 still produces the same error

**Comment by web-dave** [comment]
BrowserModule can't be found.
I'm still investigating.

**Comment by LeJeanbono** [comment]
Found it @bahmutov  !
You need to import `AppComponent` after `core-js` and `zone.js` =>
```
import 'core-js/es6/reflect';
import 'core-js/es7/reflect';
import 'zone.js/dist/zone';
import { AppComponent } from './app.component';
import { ApplicationRef, NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
```

**Comment by web-dave** [comment]
Damn, I haven't expect that.

**Comment by LeJeanbono** [comment]
![image](https://user-images.githubusercontent.com/11347342/81342631-d4165300-90b3-11ea-8f6a-669a33b05b93.png)

**Comment by LeJeanbono** [comment]
Ok, got this : 
```typescript
export class AppComponent {
  @Input() myInput;
  title = 'angular-cypress-unit';
}
```
```html
<div style="text-align:center">
  <h1>
    Welcome to {{ myInput }}!
  </h1>
</div>
```
```typescript
platformBrowserDynamic()
        .bootstrapModule(MyTestModule)
        .then(function (moduleRef) {
          const app = moduleRef.instance.app;
          app.bootstrap(AppComponent, el$.get(0)).instance.myInput = 'WORKS';
          app.tick();
        });
```
![image](https://user-images.githubusercontent.com/11347342/81344534-3755b480-90b7-11ea-88a0-f1883ff87b8e.png)

**Comment by LeJeanbono** [comment]
``` typescript
const bootComponent = (component: Type<any>, inputs: object) => {
  cy.get('app-root').then(el$ => {
    platformBrowserDynamic().bootstrapModule(MyTestModule).then(function (moduleRef) {
      const app = moduleRef.instance.app;
      const componentRef = app.bootstrap(component, el$.get(0));
      Object.keys(inputs).forEach(inputName => {
        componentRef.instance[inputName] = inputs[inputName];
      });
      app.tick();
    });
  });
};
```
Spec file :
```javascript
it('works', () => {
    bootComponent(AppComponent, {myInput: 'WORKS'});
    cy.contains('TUTU Welcome to WORKS');
  });

  it('works 2', () => {
    bootComponent(AppComponent, {myInput: 'WORKS', other: 'Bonjour'});
    cy.contains('Bonjour Welcome to WORKS');
  });
```

**Comment by bahmutov** [comment]
@LeJeanbono @web-dave I have released v2.0.0 of this plugin and am trying to apply it to Angular9 project, but hit a blocker https://github.com/bahmutov/ng9-tour-of-heroes/issues/1 - how do you set up your Webpack transpiler in that case? Right now I load it myself using old dependencies like angular2-template-loader but I really would like to point at existing Webpack config by asking Angular CLI /dev kit and using it. This is what I want

```
const webpackConfig = require('some angular dependency).getWebpack()
// pass webpack to Cypress webpack preprocessor
```

Then the components and specs would be bundled the same way as the application and all would be 100% perfect
